### ISSUE TITLE
Using ProMix with Clean Datasets – Avoiding NaN Loss

### ISSUE THREAD

**Issue by yujinlee815** [Justherozen/ProMix#18]
If you're looking to use the ProMix algorithm with a clean dataset, please read this issue carefully.

When using ProMix on clean datasets, there is a possibility that, after a certain number of training steps, idx_chosen, idx_chosen2, idx_unchosen, and idx_unchosen2 may become empty. This can result in NaN values in the loss calculation since there are no instances selected.

To avoid this issue, make sure to compute the loss only when these indices contain elements. You can do this by adding a conditional check before calculating the loss.

For example,
if idx_unchosen.numel():
    loss_unrel_ph = CEsoft(outputs_x_unrel_ph[idx_unchosen], targets=targets_urel[idx_unchosen]).mean() \
                    + w * CEsoft(outputs_x2_unrel_ph[idx_unchosen], targets=targets_urel[idx_unchosen]).mean()

**Comment by hbzju** [comment]
Thanks for pointing this out!
